src/c++/perf_analyzer/custom_load_manager.cc

```diff
@@ -76,24 +76,60 @@ CustomLoadManager::CustomLoadManager(
 cb::Error
 CustomLoadManager::InitCustomIntervals()
 {
-  schedule_.clear();
-  schedule_.emplace_back(0);
-  if (!request_intervals_file_.empty()) {
-    RETURN_IF_ERROR(
-        ReadTimeIntervalsFile(request_intervals_file_, &custom_intervals_));
-    size_t index = 0;
-    while (schedule_.back() < *gen_duration_) {
-      std::chrono::nanoseconds next_timestamp(
-          schedule_.back() + custom_intervals_[index++]);
-      schedule_.emplace_back(next_timestamp);
-      if (index == custom_intervals_.size()) {
-        index = 0;
-      }
-    }
+  PauseWorkers();
+  auto status = GenerateSchedule();
+  ResumeWorkers();
+  return status;
+}
+
+cb::Error
+CustomLoadManager::GenerateSchedule()
+{
+  if (request_intervals_file_.empty()) {
+    return cb::Error::Success;
   }
+
+  RETURN_IF_ERROR(
+      ReadTimeIntervalsFile(request_intervals_file_, &custom_intervals_));
+
+  auto worker_schedules = CreateWorkerSchedules();
+  GiveSchedulesToWorkers(worker_schedules);
   return cb::Error::Success;
 }
 
+std::vector<RateSchedulePtr_t>
```

Comment:
This is all still relying on copy elision to not copy the vector, which _usually_ happens but not always. I recommend passing in an output parameter vector reference or returning a shared_ptr to a vector.

Reply:
I confirmed the vector pointer is the same all the way up and down the chain.